source/extensions/common/wasm/wasm.cc

```diff
@@ -1473,26 +1533,129 @@ void Context::setEncoderFilterCallbacks(Envoy::Http::StreamEncoderFilterCallback
   encoder_callbacks_ = &callbacks;
 }
 
-void Context::onAsyncClientSuccess(uint32_t token, Envoy::Http::MessagePtr& response) {
+void Context::onHttpCallSuccess(uint32_t token, Envoy::Http::MessagePtr& response) {
   auto body =
       absl::string_view(static_cast<char*>(response->body()->linearize(response->body()->length())),
                         response->body()->length());
-  onHttpCallResponse(token, getHeaderPairs(&response->headers()), body,
-                     getHeaderPairs(response->trailers()));
+  onHttpCallResponse(token, headerMapToPairs(&response->headers()), body,
+                     headerMapToPairs(response->trailers()));
   http_request_.erase(token);
 }
 
-void Context::onAsyncClientFailure(uint32_t token, Http::AsyncClient::FailureReason /* reason */) {
+void Context::onHttpCallFailure(uint32_t token, Http::AsyncClient::FailureReason /* reason */) {
   onHttpCallResponse(token, {}, "", {});
   http_request_.erase(token);
 }
 
 void AsyncClientHandler::onSuccess(Envoy::Http::MessagePtr&& response) {
-  context->onAsyncClientSuccess(token, response);
+  context->onHttpCallSuccess(token, response);
 }
 
 void AsyncClientHandler::onFailure(Http::AsyncClient::FailureReason reason) {
-  context->onAsyncClientFailure(token, reason);
+  context->onHttpCallFailure(token, reason);
+}
+
+void GrpcCallClientHandler::onCreateInitialMetadata(Http::HeaderMap& metadata) {
+  context->onGrpcCreateInitialMetadata(token, metadata);
```

Comment:
This is another place that causes a weird crashing. Printf debugging tells me that crash happens when call is made into 'onGrpcCreateInitialMetadata'. I checked there is no nullptr problem and I am out of idea what could be wrong here.. Here is stack trace:

```
unhandled SIGSEGV
Call stack:
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!/lib/x86_64-linux-gnu/libpthread.so.0!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  <unknown function>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!./bazel-bin/source/exe/envoy-static!<unknown>
  host!/lib/x86_64-linux-gnu/libpthread.so.0!<unknown>
  host!/lib/x86_64-linux-gnu/libc.so.6!clone+63
[2019-04-17 13:27:59.139][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:81] Caught Aborted, suspect faulting address 0x7ebe
[2019-04-17 13:27:59.139][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:67] Backtrace (use tools/stack_decode.py to get line numbers):
[2019-04-17 13:27:59.159][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #0: Envoy::SignalAction::sigHandler() [0x563b94407ac4]
[2019-04-17 13:27:59.159][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #1: __restore_rt [0x7fa9e1e4c0c0]
[2019-04-17 13:27:59.176][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #2: WAVM::Platform::getPageSizeLog2() [0x563b927aa290]
[2019-04-17 13:27:59.193][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #3: signalHandler() [0x563b927a9f8f]
[2019-04-17 13:27:59.193][32491][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:73] #4: __restore_rt [0x7fa9e1e4c0c0]
Aborted
```

Not sure if it relates to using default grpcSimpleCall without customized grpccallbackhandler in wasm module.